src/crypto/Crypto.cpp

```diff
@@ -126,6 +126,19 @@ void Crypto::raiseError(const QString& str)
     qWarning("Crypto::selfTest: %s", qPrintable(m_errorStr));
 }
 
+bool Crypto::testSha1()
+{
+    QByteArray sha1Test =
+        CryptoHash::hash("abcdbcdecdefdefgefghfghighijhijkijkljklmklmnlmnomnopnopq", CryptoHash::Sha1);
+
+    if (sha1Test != QByteArray::fromHex("84983E441C3BD26EBAAE4AA1F95129E5E54670F1")) {
```

Comment:
Is this a reference hash from the specification or this or is it just a hash you generated with another tool? You should use and reference the former if possible.